### ISSUE TITLE
Escape reserved characters

### ISSUE THREAD

**Issue by intranets-talk** [sleimanx2/plastic#86]
Hi,
I've managed to set up Plastic as per documentation info - and works fine, thank you.

I'm having a issue where using ES "reserved characters" in queries throws a BadRequest400Exception  - query_shard_exception: Failed to parse query. Example: searching for **test:**, throws out the mentioned exception.

I am aware that this is more of a ES related issue, not Plastic's but I was wondering how others managed to get around this. Is query sanitation required, or do I need to use some regex? I know I can catch the exception, but I rather not.

Example of how I am building the query: 

```
Attachment::search()->querystring('*' . $query . '*' , ['fields' => ['name^4', 'file_slug', 'path', 'extension', 'size']])->size(15)->get()->hits();     
```

The ES reserved characters are listed here: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters

![2017-04-03_10-51](https://cloud.githubusercontent.com/assets/1742532/24620848/855d1d02-185c-11e7-93fd-e531ea7d0beb.jpg)

**Comment by sleimanx2** [comment]
Yes as per the documentation not escaping the reserved characters could lead to a syntax error. I'm still not sure if escaping those characters automatically is a good idea. Do you have any insights?

**Comment by intranets-talk** [comment]
Agree, escaping the reserved characters it's probably not the way to go, and playing with this can render inaccurate search results. 

What I am doing right now is allowing some of the characters (which I believe are "most used"), like **+ : - &** - by using **preg_replace** . As for the other characters, they only throw the exception if a wildcard is used in query. This takes away some of the ES functionality, but I have nothing better so far. At least this way, I get accurate results.

I guess this can be closed, is it not as issue with Plastic, but a request for info/enhancement.
Thanks again.
